src/components/dialogs/network-modifications/generator/modification/generator-modification-dialog.tsx

```diff
@@ -306,7 +372,7 @@ const GeneratorModificationDialog = ({
     }, [generatorToModify]);
 
     const onSubmit = useCallback(
-        (generator) => {
+        (generator: GeneratorModificationSchemaForm) => {
```

Comment:
in submit, should post an object with type conform to one in the back, do toModificationOperation() if needed, do not transform again in the api call which can lead to error..